src/js/plugins/tts/AbstractTTSEngine.js

```diff
@@ -143,4 +204,55 @@ export default class AbstractTTSEngine {
             }
         );
     }
-}
\ No newline at end of file
+
+    /** Convenience wrapper for {@see AbstractTTSEngine.getBestVoice} */
+    getBestVoice() {
+        return AbstractTTSEngine.getBestBookVoice(this.getVoices(), this.opts.bookLanguage);
+    }
+
+    /**
+     * @private
+     * Find the best voice to use given the available voices, the book language, and the user's
+     * languages.
+     * @param {SpeechSynthesisVoice[]} voices
+     * @param {ISO6391} bookLanguage
+     * @param {string[]} userLanguages languages in BCP47 format (e.g. en-US). Ordered by preference.
+     * @return {SpeechSynthesisVoice | undefined}
+     */
+    static getBestBookVoice(voices, bookLanguage, userLanguages=navigator.languages) {
+        const bookLangVoices = voices.filter(v => v.lang.startsWith(bookLanguage));
+        // navigator.languages browser support isn't great yet, so get just 1 langauge otherwise
+        // Sample navigator.languages: ["en-CA", "fr-CA", "fr", "en-US", "en", "de-DE", "de"]
+        userLanguages = userLanguages || (navigator.language ? [navigator.language] : []);
+
+        // user languages that match the book language
```

Comment:
findVoiceMatchingLanguageAndBrowserLocale(voices, language, browserLocales)
- voicesInLanguage = getVoicesInLanguage(voices, language)
- voicesInLocale = getVoicesInLocale(voices, browserLocales)
- (intersect(voicesInLanguage, voicesInLocale) || voicesInLanguage)[0]

Reply:
Doing these changes in a future PR